src/core/symbology-ng/qgspointdisplacementrenderer.cpp

```diff
@@ -301,6 +308,69 @@ void QgsPointDisplacementRenderer::calculateSymbolAndLabelPositions( QgsSymbolRe
       }
       break;
     }
+    case Grid:
+    {
+      double centerDiagonal = symbolContext.renderContext().convertToPainterUnits( M_SQRT2 * mCenterSymbol->size(),
+                              mCenterSymbol->sizeUnit(), mCenterSymbol->sizeMapUnitScale() );
+      int pointsRemaining = nPosition;
+      gridSize = ceil( sqrt( pointsRemaining ) );
+      if ( pointsRemaining - pow( gridSize - 1, 2 ) < gridSize )
+        gridSize -= 1;
+      double pointRadius = ( centerDiagonal / 2.0 + symbolDiagonal / 2.0 ) + symbolDiagonal;
```

Comment:
I'd suggest:

- Change this line to:

      double pointRadius = ( centerDiagonal / 2.0 + symbolDiagonal / 2.0 ) + symbolDiagonal + circleAdditionPainterUnits;

- Change the label for "Ring size adjustment" in the ui to just "Size adjustment"

Then users can tweak the grid spacing as desired. We'll never be able to autocalculate an ideal spacing which works for all symbol types, so it's a good idea to give this control to the user.